app/models/order.rb

```diff
@@ -0,0 +1,31 @@
+class Order < ApplicationRecord
+  enum :status, {'In Progress': 0, 'Placed': 1, 'Delivered': 2, 'Cancelled': 3}
+  
+  belongs_to :user
+  belongs_to :address, optional: true
+  has_many :payments
+  has_many :line_items, dependent: :destroy
+  has_many :deals, through: :line_items
+
+  scope :placed_orders, ->{ includes(:address, :payments).where.not(status: 'In Progress').where(payments: { status: 'Successful' }) }
```

Comment:
placed orders should just be those whose status is placed

Reply:
fixed